src/tree/ingest.rs

```diff
@@ -28,6 +31,45 @@ impl<'a> Ingestion<'a> {
     ///
     /// Will return `Err` if an IO error occurs.
     pub fn new(tree: &'a Tree) -> crate::Result<Self> {
+        // Block concurrent writes and concurrent ingestions for the duration of this ingestion.
+        let ingestion_guard = tree.0.ingestion_lock.write().expect("lock is poisoned");
```

Comment:
With the new ingestion API, I think we don't actually need this lock anymore!
Because memtable rotations and setting up flushes block out writes, once a flush process starts, we can continue writing into the active memtable because it cannot sneak past the ongoing flush.

Reply:
The revamped flush API is merged now into main